src/transports/stun.rs

```diff
@@ -0,0 +1,531 @@
+            return Err(StunError::InvalidResponse(
+                "Message length mismatch".to_string(),
+            ));
+        }
+
+        // Parse attributes
+        let mut offset = STUN_HEADER_SIZE;
+        let mut xor_mapped_addr: Option<SocketAddr> = None;
+        let mut mapped_addr: Option<SocketAddr> = None;
+
+        while offset + 4 <= STUN_HEADER_SIZE + msg_len {
+            let attr_type = BigEndian::read_u16(&data[offset..offset + 2]);
+            let attr_len = BigEndian::read_u16(&data[offset + 2..offset + 4]) as usize;
+
+            if offset + 4 + attr_len > data.len() {
+                break;
+            }
+
+            let attr_data = &data[offset + 4..offset + 4 + attr_len];
+
+            match attr_type {
+                ATTR_XOR_MAPPED_ADDRESS => {
+                    xor_mapped_addr = Self::parse_xor_mapped_address(attr_data, &data[4..8]);
+                }
+                ATTR_MAPPED_ADDRESS => {
+                    mapped_addr = Self::parse_mapped_address(attr_data);
+                }
+                _ => {
+                    trace!("Ignoring STUN attribute 0x{:04x}", attr_type);
+                }
+            }
+
+            // Attributes are padded to 4-byte boundaries
+            offset += 4 + ((attr_len + 3) & !3);
+        }
+
+        // Prefer XOR-MAPPED-ADDRESS over MAPPED-ADDRESS
+        xor_mapped_addr
+            .or(mapped_addr)
+            .ok_or_else(|| StunError::InvalidResponse("No mapped address in response".to_string()))
+    }
+
+    /// Parses XOR-MAPPED-ADDRESS attribute
+    fn parse_xor_mapped_address(data: &[u8], magic_cookie: &[u8]) -> Option<SocketAddr> {
+        if data.len() < 8 {
+            return None;
+        }
+
+        let family = data[1];
+        let xor_port = BigEndian::read_u16(&data[2..4]);
+        let port = xor_port ^ ((STUN_MAGIC_COOKIE >> 16) as u16);
+
+        match family {
+            0x01 => {
+                // IPv4
+                if data.len() < 8 {
+                    return None;
+                }
+                let xor_addr = BigEndian::read_u32(&data[4..8]);
+                let addr = xor_addr ^ STUN_MAGIC_COOKIE;
+                let ip = std::net::Ipv4Addr::from(addr);
+                Some(SocketAddr::new(ip.into(), port))
+            }
+            0x02 => {
+                // IPv6
+                if data.len() < 20 {
+                    return None;
+                }
+                let mut addr_bytes = [0u8; 16];
+                addr_bytes.copy_from_slice(&data[4..20]);
+                // XOR with magic cookie (first 4 bytes) and transaction ID (last 12 bytes)
+                for i in 0..4 {
+                    addr_bytes[i] ^= magic_cookie[i];
+                }
+                // Note: For full compliance, we'd need transaction ID here
+                // This is a simplified implementation
+                let ip = std::net::Ipv6Addr::from(addr_bytes);
```

Comment:
why not just add the transaction ID here? It's a pretty simple addition and we are closer to being compliant to the spec in a sensible way